snuba/subscriptions/data.py

```diff
@@ -95,7 +104,7 @@ def from_string(cls, value: str) -> SubscriptionIdentifier:
 
 
 @dataclass(frozen=True, kw_only=True)
-class SubscriptionData(ABC):
+class _SubscriptionData(ABC, Generic[TRequest]):
```

Comment:
`build_request` returns generic type `TRequest` and `run_query` accepts `TRequest`.
So for SnQLSubscriptionData, `build_request` returns `Request` and `run_query` accepts `Request`
and for RPCSubscriptionData, `build_request` returns `TimeSeriesRequest` and `run_query` accepts `TimeSeriesRequest`